source/MaterialXRuntime/Commands/PrimCommands.cpp

```diff
@@ -35,6 +35,27 @@ void createPrim(RtStagePtr stage, const RtToken& typeName, const RtPath& path, R
 
 void createPrim(RtStagePtr stage, const RtToken& typeName, const RtPath& parentPath, const RtToken& name, RtCommandResult& result)
 {
+    static RtTokenSet LOOK_MANAGEMENT_NODES { RtToken("collection"),
+                                              RtToken("look"),
+                                              RtToken("lookgroup"),
+                                              RtToken("materialassign")
+                                            };
+    static RtTokenSet MATERIAL_NODES { RtToken("surfacematerial"),
+                                       RtToken("volumematerial")
+                                     };
+    RtPrim nodeDefPrim =
+        RtApi::get().getMasterPrim(RtToken(typeName));
+
+    // If the node does not have a NodeDef it's either a look management node or a material node
```

Comment:
Should add in nodegraphs into nodgraphs as well which should be prevented here. 

Guess we can also prune top level inputs and outputs here, and the message can be the same for all types: THat it's invalid to create a child name X under a parent path Y.


Sure there are other situations.